tests/benchmark/ExecutorBench.cpp

```diff
@@ -0,0 +1,395 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Backends/DeviceManager.h"
+#include "glow/Runtime/Executor/Executor.h"
+
+#include "ExecutorBenchmarkWrapper.h"
+#include "ExecutorTestBuilder.h"
+#include "TestDeviceManager.h"
+
+#include "benchmark/benchmark.h"
+
+using namespace glow;
+using namespace glow::runtime;
+
+/// Fixture class for reusing Executor, ExecutorTestBuilder and DeviceManagerMap
+/// setup across benchmarks.
+class ThreadPoolExecutorBench : public benchmark::Fixture {
+public:
+  ThreadPoolExecutorBench()
+      : executor_(std::shared_ptr<Executor>(
+            createExecutor(deviceManagerMap_, ExecutorKind::ThreadPool))),
+        benchmarkBuilder_(executor_, deviceManagerMap_) {}
+  ~ThreadPoolExecutorBench() = default;
+
+protected:
+  /// The Executor being tested.
+  std::shared_ptr<Executor> executor_;
+  /// An ExecutorTestBuilder instance for creating benchmarks.
+  ExecutorTestBuilder benchmarkBuilder_;
+  /// DeviceManager map for initializing executor_.
+  DeviceManagerMapTy deviceManagerMap_;
+};
+
+/// Benchmark for executing an empty DAG.
+BENCHMARK_DEFINE_F(ThreadPoolExecutorBench, Empty)(benchmark::State &state) {
+  constexpr RunIdentifierTy testRunId = 10;
+
+  for (auto _ : state) {
```

Comment:
I'm not familiar enough with gbench I guess, what are we iterating over here?

Reply:
The different iterations of the benchmark. If you pass a range of arguments, you can access the specific value the benchmark should run with through `state`.